src/fw-update/fw-update-device-interface.h

```diff
@@ -15,6 +15,20 @@ namespace librealsense
         virtual std::vector<uint8_t> backup_flash(update_progress_callback_ptr callback) = 0;
         virtual void update_flash(const std::vector<uint8_t>& image, update_progress_callback_ptr callback, int update_mode) = 0;
         virtual bool check_fw_compatibility(const std::vector<uint8_t>& image) const = 0;
+        std::string extract_firmware_version_string(const void* fw_image, size_t fw_image_size) const
+        {
+            uint32_t version{};
+
+            memcpy(reinterpret_cast<char*>(&version), reinterpret_cast<const char*>(fw_image) +
```

Comment:
Validate size and buffer ptr before dereferencing

Reply:
Done